packages/playwright/src/reporters/merge.ts

```diff
@@ -81,19 +81,37 @@ const commonEvents = new Set(commonEventNames);
 const commonEventRegex = new RegExp(`${commonEventNames.join('|')}`);
 
 function parseCommonEvents(reportJsonl: Buffer): JsonEvent[] {
-  return reportJsonl.toString().split('\n')
+  return splitBufferLines(reportJsonl)
+      .map(line => line.toString('utf8'))
       .filter(line => commonEventRegex.test(line)) // quick filter
       .map(line => JSON.parse(line) as JsonEvent)
       .filter(event => commonEvents.has(event.method));
 }
 
 function parseTestEvents(reportJsonl: Buffer): JsonEvent[] {
-  return reportJsonl.toString().split('\n')
+  return splitBufferLines(reportJsonl)
+      .map(line => line.toString('utf8'))
       .filter(line => line.length)
       .map(line => JSON.parse(line) as JsonEvent)
       .filter(event => !commonEvents.has(event.method));
 }
 
+function splitBufferLines(buffer: Buffer) {
+  const lines = [];
+  let start = 0;
+  while (start < buffer.length) {
```

Comment:
Maybe we should do something like this instead?

```ts
    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    for await (const line of rl) {
        ...
    }
```

but probably not a huge difference if at the end the lines are anyways in the memory.

Reply:
Yeah, the buffer is in memory anyway, so I'd keep it simple for now.